arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1350,8 +1350,14 @@ int ClusterInfo::createCollectionCoordinator(std::string const& databaseName,
     // Update our cache:
     loadPlan();
   }
+      
+  bool isSmart = false;
+  VPackSlice smartSlice = json.get("isSmart");
+  if (smartSlice.isBool() && smartSlice.getBool()) {
+    isSmart = true;
+  }
 
-  if (numberOfShards == 0) {
+  if (numberOfShards == 0 || isSmart) {
```

Comment:
It's okay like this.
We need to check if numberOfShards == 0 is superfluous. (Is afaik only allowed on Smart)